src/app/core/entity/schema/entity-schema.service.ts

```diff
@@ -140,12 +150,24 @@ export class EntitySchemaService {
     const data = {};
 
     for (const key of schema.keys()) {
+      let value = entity[key];
       const schemaField: EntitySchemaField = schema.get(key);
 
-      if (entity[key] !== undefined) {
-        data[key] = this.getDatatypeOrDefault(
-          schemaField.dataType
-        ).transformToDatabaseFormat(entity[key], schemaField, this, entity);
+      if (value === undefined) {
+        if (schemaField.defaultValue !== undefined) {
+          value = schemaField.defaultValue;
+        } else {
+          // skip and keep undefined
+          continue;
```

Comment:
And shouldn't then the undefined checks be removed from the datatype definitions for clarity?

Reply:
agreed, will remove them in the individual data-type implementations